roscala/src/main/scala/coop/rchain/rosette/prim/Number.scala

```diff
@@ -81,21 +81,418 @@ object Number {
     override val minArgs: Int = 2
     override val maxArgs: Int = 2
 
+    @checkTypeMismatch[Fixnum]
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Fixnum] = {
+      val m = ctxt.argvec.elem.head.asInstanceOf[Fixnum]
+      val n = ctxt.argvec.elem(1).asInstanceOf[Fixnum]
+
+      try {
+        Right(m % n)
+      } catch {
+        case e: ArithmeticException =>
+          Left(ArithmeticError)
+      }
+    }
+  }
+
+  object fxLt extends Prim {
+    override val name: String = "fx<"
+    override val minArgs: Int = 2
+    override val maxArgs: Int = 2
+
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, RblBool] =
+      checkFixnum(0, ctxt.argvec.elem) match {
+        case Left(m) => Left(m)
```

Comment:
Please do this instead:
```
checkFixnum(0, ctxt.argvec.elem).map{ m =>
   [...]
}

Reply:
Yes, you're right. I have changed the code as you said. It really makes the code more clear.